tests/dataset/measurement/test_measurement_context_manager.py

```diff
@@ -201,6 +207,23 @@ def test_register_custom_parameter(DAC) -> None:
         )
 
 
+def test_register_delegate_parameters():
+    x_param = Parameter("x", set_cmd=None, get_cmd=None)
+
+    complex_param = Parameter(
+        "complex_param", get_cmd=None, set_cmd=None, vals=ComplexNumbers()
+    )
+    delegate_param = DelegateParameter("delegate", source=complex_param)
```

Comment:
The case that failed previously was when the delegate parameter was first created with no source and then appointed a source afterwards, so I suggest to test that case. Unless it is covered by the other new tests and does not need to be part of this test.

```suggestion
    delegate_param = DelegateParameter("delegate", source=None)
    delegate_param.source = complex_param
```

Reply:
Extended with another test that covers this